frontend/public/components/factory/list-page.jsx

```diff
@@ -57,6 +59,14 @@ TextFilter.displayName = 'TextFilter';
 
 /** @augments {React.PureComponent<{ListComponent: React.ComponentType<any>, kinds: string[], flatten?: function, data?: any[], rowFilters?: any[]}>} */
 export class ListPageWrapper_ extends React.PureComponent {
+  componentDidMount() {
+    if (!_.isEmpty(this.props.kinds) && _.size(this.props.kinds) === 1) {
+      store.dispatch(UIActions.setTableObject(kindObj(this.props.kinds[0])));
```

Comment:
I need to learn more about redux, but this doesn't seem right. If there is ever more than one table on a page will this fail?

Reply:
I dont think it will fail but nothing will be rendered since the action for getting the table wont be triggered.